FlowCrypt/Controllers/Setup/SetupViewController.swift

```diff
@@ -174,12 +174,16 @@ extension SetupViewController {
             try await(self.alertAndSkipOnRejection(AttesterApi.shared.updateKey(email: userId.email, pubkey: encryptedPrv.key.public), fail: "Failed to submit Public Key"))
             try await(self.alertAndSkipOnRejection(AttesterApi.shared.testWelcome(email: userId.email, pubkey: encryptedPrv.key.public), fail: "Failed to send you welcome email"))
         }.then(on: .main) { [weak self] in
-            self?.performSegue(withIdentifier: "InboxSegue", sender: nil)
+            self?.moveToMainFlow()
         }.catch(on: .main) { [weak self] error in
             self?.showAlert(error: error, message: "Could not finish setup, please try again")
         }
     }
 
+    private func moveToMainFlow() {
+        GlobalRouter().proceedAfterLogOut()
```

Comment:
`proceedAfterLogout` is not the best name considering how we use it. Maybe `GlobalRouter().reloadApp()` ?